frontend/src/Game.js

```diff
@@ -9,13 +9,16 @@ export default class Game {
         this.gameMap = gameMap;
         this.over = false;
 
-        this.towers = [];
+        this.towers = new Array(gameMap.depth).fill(null).map(() => new Array(gameMap.width).fill(null));
+        // game.towers is 2D, structures in GameCanvas will be flattened for rendering
```

Comment:
instead of this.towers having all the towers, e.g., [tower1, tower2...], this makes it a 2d array containing all the towers by location in the grid, like `this.towers = [
[tower1, null], 
[null, tower2]
]`
That makes it quick to check if there's a tower at a location already.